File: GitToolBox/src/main/java/zielu/gittoolbox/ui/blame/BlameEditorData.java
package zielu.gittoolbox.ui.blame;

import com.intellij.openapi.util.Key;
import org.jetbrains.annotations.NotNull;
import zielu.gittoolbox.revision.RevisionInfo;

class BlameEditorData {
  static final Key<BlameEditorData> KEY = new Key<>("GitToolBox-blame-editor");

  private final int editorLineIndex;
  private final boolean lineModified;
  private final int generation;
  private final RevisionInfo revisionInfo;

  BlameEditorData(int editorLineIndex, boolean lineModified, int generation, @NotNull RevisionInfo revisionInfo) {
    this.editorLineIndex = editorLineIndex;
    this.lineModified = lineModified;
    this.generation = generation;
    this.revisionInfo = revisionInfo;
  }

  boolean isSameEditorLineIndex(int editorLine) {
    return this.editorLineIndex == editorLine;
  }

  boolean isSameGeneration(int generation) {
    return this.generation == generation;
  }

  boolean isLineModified() {
    return lineModified;
  }

  @NotNull
  RevisionInfo getRevisionInfo() {
    return revisionInfo;
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/ui/blame/BlameEditorData.kt
package zielu.gittoolbox.ui.blame

import com.intellij.openapi.util.Key
import com.intellij.openapi.util.UserDataHolder
import zielu.gittoolbox.revision.RevisionInfo

internal data class BlameEditorData(
  val editorLineIndex: Int,
  val lineModified: Boolean,
  val generation: Int,
  val revisionInfo: RevisionInfo
) {

  fun isSameEditorLineIndex(editorLineIndex: Int): Boolean {
    return this.editorLineIndex == editorLineIndex
  }

  fun isSameGeneration(generation: Int): Boolean {
    return this.generation == generation
  }

  companion object {
    private val KEY = Key<BlameEditorData>("GitToolBox-blame-editor")

    @JvmStatic
    fun get(userDataHolder: UserDataHolder): BlameEditorData? {
      return userDataHolder.getUserData(KEY)
    }

    @JvmStatic
    fun set(userDataHolder: UserDataHolder, editorData: BlameEditorData) {
      userDataHolder.putUserData(KEY, editorData)
    }

    @JvmStatic
    fun clear(userDataHolder: UserDataHolder) {
      userDataHolder.putUserData(KEY, null)
    }
  }
}
